app/services/catalog_remediator.rb

```diff
@@ -0,0 +1,61 @@
+# frozen_string_literal: true
+
+# Remediates catalog entries, e.g., failed replication records
+class CatalogRemediator
```

Comment:
This class is entirely uncovered right now! I'd like feedback on the approach before writing unit tests, if that's OK, just in case we want to make substantial changes.